### PR TITLE
Wallet - Create Foundry Output and Mint Native Tokens

### PR DESCRIPTION
**Pull request by matjazv**
# Description of change

- implemented #444 
- fixed serialization/deserialization of native tokens in Client module
- fixed serialization/deserialization of token scheme in Client module
- implemented calculating Foundry ID
- made `calc_minimum_output_deposit` function to be public
- implemented creating `uint256_t` number from hex string and converting it to a hex string

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

All local and newly created unit tests passed.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: examples/wallet/mint_native_tokens.c [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    printf("Get state controller address failed\n");
+    return -1;
+  }
+  if (wallet_ed25519_address_from_index(w, false, govern_addr_index, &govern_addr) != 0) {
+    printf("Get governor address failed!\n");
+    return -1;
+  }
+  if (wallet_ed25519_address_from_index(w, false, receiver_addr_index, &receiver_addr) != 0) {
+    printf("Get receiver address failed\n");
+    return -1;
```
Comment (oopsmonk) [inline examples/wallet/mint_native_tokens.c]: the wallet instance needs to be freed.

```suggestion
    printf("Get sender address failed\n");
    wallet_destroy(w);
    return -1;
  }
  if (wallet_ed25519_address_from_index(w, false, state_ctrl_addr_index, &state_ctrl_addr) != 0) {
    printf("Get state controller address failed\n");
    wallet_destroy(w);
    return -1;
  }
  if (wallet_ed25519_address_from_index(w, false, govern_addr_index, &govern_addr) != 0) {
    printf("Get governor address failed!\n");
    wallet_destroy(w);
    return -1;
  }
  if (wallet_ed25519_address_from_index(w, false, receiver_addr_index, &receiver_addr) != 0) {
    printf("Get receiver address failed\n");
    wallet_destroy(w);
    return -1;
```

File: src/wallet/output_foundry.c [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  if (!output_foundry) {
+    printf("[%s:%d] can not create foundry output\n", __func__, __LINE__);
+    ret = -1;
+    goto end;
+  }
+
+  // calculate minimum storage deposit for Foundry Output and update output's amount
+  uint64_t min_storage_deposit = calc_minimum_output_deposit(&w->byte_cost, OUTPUT_FOUNDRY, output_foundry);
+  output_foundry->amount = min_storage_deposit;
+  printf("Storage deposit for foundry output: %" PRIu64 "\n", output_foundry->amount);
```
Comment (oopsmonk) [inline src/wallet/output_foundry.c]: I think it's for debugging, should comment out in most of case.
```suggestion
  // printf("Storage deposit for foundry output: %" PRIu64 "\n", output_foundry->amount);
```

**Review by oopsmonk — CHANGES_REQUESTED** [review]
LGTM, needs some minor changes.

File: examples/wallet/mint_native_tokens.c (hunk 1 above)
Comment (matjazv) [reply examples/wallet/mint_native_tokens.c]: Thanks for catching this bug.
Done.

File: src/wallet/output_foundry.c (hunk 2 above)
Comment (matjazv) [reply src/wallet/output_foundry.c]: Debug outputs are commented out.

**Review by oopsmonk — APPROVED** [review]

**Pull request merged**